File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/data/repository/NotificationRepository.java
package com.github.arhor.simple.expense.tracker.data.repository;

import java.util.Collection;
import java.util.List;
import java.util.UUID;

import org.springframework.data.repository.CrudRepository;

import com.github.arhor.simple.expense.tracker.data.model.Notification;

public interface NotificationRepository extends CrudRepository<Notification, UUID> {

    List<Notification.CompactProjection> findAllByTargetUserIdIn(Collection<Long> targetUserIds);
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/data/repository/NotificationRepository.kt
package com.github.arhor.simple.expense.tracker.data.repository

import com.github.arhor.simple.expense.tracker.data.model.Notification
import com.github.arhor.simple.expense.tracker.data.model.projection.CompactNotificationProjection
import org.springframework.data.jdbc.repository.query.Query
import org.springframework.data.repository.CrudRepository
import java.util.UUID
import java.util.stream.Stream

interface NotificationRepository : CrudRepository<Notification, UUID> {

    @Query(name = "Notification.findAllByTargetingIdIn")
    fun findAllByTargetUserIdIn(targetUserIds: Collection<Long>): Stream<CompactNotificationProjection>
}
